### DISCUSSION TITLE
Cannot get output: 'standalone' to work with createServer

### DISCUSSION THREAD

**Discussion by niteshbalusu11** [vercel/next.js#39691] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.16.0
      npm: 8.6.0
      Yarn: 1.22.19
      pnpm: 6.11.0
    Relevant packages:
      next: 12.2.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

warn  - Latest canary version not detected, detected: "12.2.4", newest: "12.2.6-canary.0".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue

### Describe the Bug

```
module.exports = {
  output: 'standalone',
}
```
Using this `output:standalone` does not seem to work when using createServer.
Even though building the app with `next build` outputs a `.next/standalone`, next does not seem to be using it in production and still expects a node_modules folder with all modules required for the client side.

Here's my server config:
```
    try {
      this.server = createServer({
        dev: this.configService.get<string>('NODE_ENV') !== 'production',
        dir: './src/client',
      });
      await this.server.prepare();
    } catch (error) {
      console.error(error);
    }
```

### Expected Behavior

`output: standalone` uses `.next/standalone` instead of node_modules in production.

### Link to reproduction

https://github.com/niteshbalusu11/lndboss

### To Reproduce

- clone the repo
- install deps using `yarn`
- update `src/client/next.config.js` and add `output: 'standalone'`
- build the app `yarn build:prod`
- uninstall some client side dependency like `@mui/material`
- start the app using `yarn start:prod`
- go to `localhost:8055` and you'll see errors because of missing dependencies.

**Comment by SukkaW** [comment]
@niteshbalusu11 Would you mind re-checking your `.next/standalone` directory?

IIRC there should be a `.next/standalone/node_modules` directory (that only contains the required files). With `.next/standalone/node_modules` you can deploy `.next/standalone` without executing `npm i` (as a dedicated `node_modules` is there already).

Also, you should not use `next start` to start a standalone server. Use `node .next/standalone/server.js` instead.

**Reply by niteshbalusu11** [reply]
Sorry, I missed this. In my situation, I don't think I can use server.js because I use a custom server setup. The only way to use standalone output is starting the server.js file?

**Reply by SukkaW** [reply]
@niteshbalusu11 You can use `server.js` directly, or you can just adopt it into your custom server! Just copy, paste, and make some modifications.
